### ISSUE TITLE
Potential problem in evaluation

### ISSUE THREAD

**Issue by omarfoq** [owkin/FLamby#71]
In the current code for Camelyon16 and Heart_Disease we use AUC_ROC as a metric to evaluate the model. We rely on  `sklearn.metrics.roc_auc_score`, which requires the vector of predictions to sum-up to one. This is an issue because our models (see for example in `Baseline` in `datasets/fed_camelyon16/model.py`) return the logits not the probabilities; this point is confirmed by the fact that, in `datasets/fed_camelyon16/loss.py` for example, we use `BCEWithLogitsLoss`. We should either apply a sigmoid / softmax activation 1) inside the model itself (e.g., `datasets/fed_camelyon16/model.py` ), 2) after computing the outputs (i.e., line 49 in `utils.py`) or 3) inside the function `metric` (e.g., in datasets/fed_camelyon16/metric.py ).

I think the third option is the most suitable, because it ensure that the functions and classes in , for example, `datasets/fed_camelyon16` is self consistent.
